lib/puppet/provider/package/gem.rb

```diff
@@ -125,9 +127,30 @@ def self.gemsplit(desc)
 
   def insync?(is)
     return false unless is && is != :absent
+    should = @resource[:ensure]
 
+    unless should =~ /,/ or should =~ Gem::Requirement::PATTERN
```

Comment:
In ruby, in conditions, `||` is used over  `or` as they behave differently

https://zverok.github.io/blog/2016-02-23-control-flow.html